### PR TITLE
Added a processor.get for visa and mastercard.

### PR DESCRIPTION
**Pull request by asnahassan**
Add processor.get() function in model-acquirer which were visa and mastercard.

### CODE REVIEW

File: Merchant/Processor.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export interface Processor {
 	}
 	ipg?: { reference: string }
 	baboba?: { reference: string }
+	clearhaus?: { reference: string }
```
Comment (robinnedergard) [inline Merchant/Processor.ts:12]: We have to have a merchant specific key, that won't be specified in the key for our acquirer:
```suggestion
	clearhaus?: { 
	reference: string
	key: string 
	}
```

**Review by robinnedergard — CHANGES_REQUESTED** [review]

File: Merchant/index.ts [hunk 2]
```diff
@@ -46,7 +46,9 @@ export namespace Merchant {
 			isoly.CountryCode.Alpha2.is(value.country) &&
 			typeof value.categoryCode == "string" &&
 			MerchantRules.is(value.rules) &&
-			(value.processor == undefined || MerchantProcessor.is(value.processor))
+			(value.processor == undefined ||
+				MerchantProcessor.is(value.processor) ||
+				(MerchantProcessor.is(value.processor.visa) && MerchantProcessor.is(value.processor.mastercard)))
```
Comment (robinnedergard) [inline Merchant/index.ts:51]: You are accessing a property (value.processor) that may not even be an object, which could result in a crash if it is a string or simular.
```suggestion
				(typeof value.processor == "object" && value.processor &&
				MerchantProcessor.is(value.processor.visa) && MerchantProcessor.is(value.processor.mastercard)))
```

File: Merchant/index.ts [hunk 3]
```diff
@@ -79,7 +81,10 @@ export namespace Merchant {
 							MerchantRules.is(value.rules) || MerchantRules.flaw(value.rules),
 							value.processor == undefined ||
 								MerchantProcessor.is(value.processor) ||
-								MerchantProcessor.flaw(value.processor),
+								(MerchantProcessor.is(value.processor.visa) && MerchantProcessor.is(value.processor.mastercard)) || {
```
Comment (robinnedergard) [inline Merchant/index.ts:84]:
```suggestion
								(MerchantProcessor.is(value.processor.visa) && MerchantProcessor.is(value.processor.mastercard)) || {
```

Comment (robinnedergard) [inline Merchant/index.ts:84]:
```suggestion
								(typeof value.processor == "object" && value.processor &&
								MerchantProcessor.is(value.processor.visa) && MerchantProcessor.is(value.processor.mastercard)) || {
```

Comment (robinnedergard) [inline Merchant/index.ts:84]:
```suggestion
								(typeof value.processor == "object" && value.processor &&
								MerchantProcessor.is(value.processor.visa) && MerchantProcessor.is(value.processor.mastercard)) || {
```

**Review by robinnedergard — CHANGES_REQUESTED** [review]

**Review by nilssonemma — APPROVED** [review]

File: Merchant/index.ts [hunk 4]
```diff
@@ -88,26 +99,37 @@ export namespace Merchant {
 	}
 	export function getReference(
 		merchant: Merchant,
-		processor: "omnipay" | "ipg" | "silverflow" | "baboba"
+		processor: "omnipay" | "ipg" | "silverflow" | "baboba",
```
Comment (Magnus93) [inline Merchant/index.ts:102]: Add case for `"clearhaus"` (`merchantProcessor?.clearhaus.reference`)

**Review by Magnus93 — CHANGES_REQUESTED** [review]

File: Merchant/index.ts (hunk 4 above)
Comment (asnahassan) [reply Merchant/index.ts:102]: Fixed

**Review by Magnus93 — APPROVED** [review]

**Review by marcushanses — APPROVED** [review]

**Review by fredrikbryntesson — APPROVED** [review]

File: Merchant/index.spec.ts [hunk 5]
```diff
@@ -45,13 +45,8 @@ describe("Rule", () => {
...
 				{
-					flaws: [
-						{
-							property: "silverflow",
-							type: "{reference: string, acceptor: Acceptor, key: string} | undefined",
-						},
-					],
-					type: "Processor",
+					property: "processor",
+					type: "undefined | Processor | { visa: Processor; mastercard: Processor } ",
```
Comment (simonmika) [inline Merchant/index.spec.ts:49]: Should use recursive flaws call.

**Pull request merged**